src/components/video-card/VideoCard.jsx

```diff
@@ -7,12 +7,67 @@ import {
 	FaShareSquare,
 	AiFillCloseCircle,
 } from "../../icons";
+import {
+	addToWatchLater,
+	addToLikedVideos,
+	removeFromWatchLater,
+	removeFromLikedVideos,
+} from "../../api/apicalls";
+import { useVideoContext } from "../../context/VideoContext";
 const VideoCard = ({ video }) => {
+	const { videoState, videoDispatch } = useVideoContext();
 	const [showDropDown, setShowDropDown] = useState(false);
-	const { title, authorImageUrl, thumbnailImageUrl, channelName } = video;
+	const { _id, title, authorImageUrl, thumbnailImageUrl, channelName } = video;
 	const toggleOptionMenu = () => {
 		setShowDropDown((prev) => !prev);
 	};
+
+	const addVideoToWatchLater = async () => {
+		const response = await addToWatchLater(video);
+		if (response.success) {
+			videoDispatch({ type: "SET_WATCHLATER", payload: response.watchlater });
+			setShowDropDown((prev) => !prev);
+		} else {
+			console.log("error");
+		}
+	};
+	const removeVideoFromWatchLater = async () => {
+		const response = await removeFromWatchLater(_id);
+		if (response.success) {
+			videoDispatch({ type: "SET_WATCHLATER", payload: response.watchlater });
+			setShowDropDown((prev) => !prev);
+		} else {
+			console.log("ERROR");
+		}
+	};
+
+	const addVideoToLikedVideo = async () => {
+		const response = await addToLikedVideos(video);
+		if (response.success) {
+			console.log(response);
```

Comment:
When you have some extra time, can replace console log statements with Toast component. For now its fine.

Reply:
I will try to change it to toast before the deadline.